server/schedule/checker/rule_checker.go

```diff
@@ -151,14 +152,31 @@ func (c *RuleChecker) fixRulePeer(region *core.RegionInfo, fit *placement.Region
 	}
 	// fix down/offline peers.
 	for _, peer := range rf.Peers {
-		if c.isDownPeer(region, peer) {
-			checkerCounter.WithLabelValues("rule_checker", "replace-down").Inc()
-			return c.replaceUnexpectRulePeer(region, rf, fit, peer, downStatus)
-		}
 		if c.isOfflinePeer(peer) {
+			if witness, ok := c.hasAvailableWitness(region, peer); ok {
+				checkerCounter.WithLabelValues("rule_checker", "promote-witness").Inc()
+				return operator.CreateNonWitnessPeerOperator("promote-witness", c.cluster, region, witness)
+			}
 			checkerCounter.WithLabelValues("rule_checker", "replace-offline").Inc()
 			return c.replaceUnexpectRulePeer(region, rf, fit, peer, offlineStatus)
 		}
+		if c.isDownPeer(region, peer) {
```

Comment:
Why change the order, the order is a kind of priority.

Reply:
Fixed, thanks!